app/Fission/Internal/Orphanage/RIO.hs

```diff
@@ -12,19 +12,137 @@ import qualified RIO.ByteString.Lazy as Lazy
 import           Servant.Client
 import           Network.HTTP.Client as HTTP
 
+import           Network.AWS
+import qualified Network.AWS.Auth as AWS
+import           Network.AWS.Route53
+
 import           Network.IPFS
 import           Network.IPFS.Types         as IPFS
 import qualified Network.IPFS.Process.Error as Process
 import           Network.IPFS.Process
 import qualified Network.IPFS.Peer as Peer
 
+import           Fission.AWS.Route53.Class
+import           Fission.AWS.Types as AWS
+import           Fission.AWS
 import qualified Fission.Config as Config
+import           Fission.Internal.UTF8
+import           Fission.IPFS.DNSLink
+
+instance (Has AWS.AccessKey cfg, Has AWS.SecretKey cfg) => MonadAWS (RIO cfg) where
+  liftAWS awsAction = do
```

Comment:
Question: Where does `liftAWS` get used?

Reply:
@bechurch 

`MonadAWS` is defined by the AWS package that we're using. It's used internally in all of the calls to AWS that we make. You won't see it used on the surface, since it's being used to hook our code into their library.

A pattern to get a lot of benefit from ad hoc polymorphism is to define some general function, and then get a bunch of functions that depend on that "for free".

For example, in Elixir ad hoc polymorphism is provided by protocols. For `Enumerable`, you define at least<sup>1</sup> `reduce/3` for you data type. Then you get access to almost all of the `Enum` module. The upshot is that for defining one function, you get several dozen that also work on your custom data type.

<sup>1</sup> `Enumerable` has 4 functions to define, but most only actually need `reduce/3`

Reply:
Nice explanation!